oidc-api/src/main/java/uk/gov/di/authentication/oidc/lambda/TokenHandler.java

```diff
@@ -267,22 +265,52 @@ public APIGatewayProxyResponseEvent handleRequest(
                                             && !clientSession
                                                     .getEffectiveVectorOfTrust()
                                                     .containsLevelOfConfidence();
-
                             final OIDCClaimsRequest finalClaimsRequest = claimsRequest;
-                            var tokenResponse =
-                                    segmentedFunctionCall(
-                                            "generateTokenResponse",
-                                            () ->
-                                                    tokenService.generateTokenResponse(
-                                                            clientID,
-                                                            new Subject(userProfile.getSubjectID()),
-                                                            authRequest.getScope(),
-                                                            additionalTokenClaims,
-                                                            publicSubject,
-                                                            vot,
-                                                            userProfile.getClientConsent(),
-                                                            isConsentRequired,
-                                                            finalClaimsRequest));
+                            OIDCTokenResponse tokenResponse;
+                            if (isDocCheckingAppUserWithSubjectId(clientSession)) {
+                                Scope scope =
+                                        new Scope(
+                                                getRequestObjectClaim(
+                                                        authRequest, "scope", String.class));
+                                tokenResponse =
+                                        segmentedFunctionCall(
+                                                "generateTokenResponse",
+                                                () ->
+                                                        tokenService.generateTokenResponse(
+                                                                clientID,
+                                                                clientSession.getDocAppSubjectId(),
+                                                                scope,
+                                                                additionalTokenClaims,
+                                                                clientSession.getDocAppSubjectId(),
+                                                                vot,
+                                                                null,
+                                                                false,
+                                                                finalClaimsRequest,
+                                                                true));
```

Comment:
Could set some of these fields as variables and then only have this method call once